chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -31,19 +86,59 @@ impl EvmState for super::Ledger {
         &self.evm.environment
     }
 
-    fn account(&self, address: Address) -> Option<Account> {
-        self.evm.accounts.get(&address).cloned()
+    fn account(&self, evm_address: EvmAddress) -> Option<EvmAccount> {
+        match self.evm.adress_mapping.jor_address(evm_address) {
+            Some(jor_address) => match self.accounts.get_state(&jor_address) {
+                Ok(account) => Some(EvmAccount {
+                    balance: account.value.0.into(),
+                    state: account.evm_state.clone(),
+                }),
+                Err(_) => None,
+            },
+            None => None,
+        }
     }
 
-    fn contains(&self, address: Address) -> bool {
-        self.evm.accounts.contains(&address)
+    fn contains(&self, address: EvmAddress) -> bool {
+        self.evm.adress_mapping.jor_address(address).is_some()
     }
 
-    fn modify_account<F>(&mut self, address: Address, f: F)
+    fn modify_account<F>(&mut self, address: EvmAddress, f: F) -> Result<(), String>
     where
-        F: FnOnce(Account) -> Option<Account>,
+        F: FnOnce(EvmAccount) -> Option<EvmAccount>,
     {
-        self.evm.accounts = self.evm.accounts.clone().modify_account(address, f);
+        let address = self.evm.adress_mapping.jor_address(address).ok_or(format!(
+            "Can not find corresponding jormungadr account for the evm account: {}",
+            address
+        ))?;
+
+        let account = self
```

Comment:
It is no appropriate to use `self.account(address)` at this place because this function returns `EvmAccount` instead of `JorAccount` which is needed at this place.

Reply:
But ultimately you need the `EvmAccount` in line 121, don't you?

Reply:
Yes, you are right, have missed that part. But the most crucial part is to treat an error in this place.
```
        let address = self
            .evm
            .address_mapping
            .jor_address(&address)
            .ok_or_else(|| {
                ExitError::Other(
                    format!(
                        "Can not find corresponding jormungadr account for the evm account: {}",
                        address
                    )
                    .into(),
                )
            })?
            .clone();

```
And it is something different that we want inside the `fn account(&self, evm_address: &EvmAddress)`